source/leuven/services/openondemand.rst

```diff
@@ -191,136 +241,324 @@ of using any of the other partitions. There are however some limitations on the
+
+  .. code-block :: bash
+
+     TOOLCHAIN='2023a'
+     DIR_VENV=${VSC_DATA}/apps/venv/${VSC_OS_LOCAL}/${VSC_ARCH_LOCAL}${VSC_ARCH_SUFFIX}/${TOOLCHAIN}/<kernel_name>
+     mkdir -p ${DIR_VENV}
+     # the line below is needed if you use 'Interactive Shell' app
+     module use /apps/leuven/${VSC_OS_LOCAL}/${VSC_ARCH_LOCAL}${VSC_ARCH_SUFFIX}/${TOOLCHAIN}/modules/all
+     module load Python/3.11.3-GCCcore-12.3.0
+     python -m venv ${DIR_VENV}
+     source ${DIR_VENV}/bin/activate
+     pip install --prefix=${DIR_VENV} ipykernel <additional packages>
+     # note that below, the "--env ..." argument is not needed
+     python -m ipykernel install --user --name <kernel_name> --display-name <kernel_name>
+
+- On the JupyterLab form, choose a partition to your liking and select the same toolchain as above.
+- Once you connect to your session, your new ``<kernel_name>`` is ready to be used.
+  To verify your setup, you can execute ``import sys; sys.executable`` in your notebook,
+  and the resulting path shall point at ``DIR_VENV`` where you installed your virtual environment.
+
+**Remarks:**
+
+- If one your package installation steps involves compiling source code, then you might only be able to
+  use your virtual environment on the same architecture where the compilation was carried out. 
+  This remark does not apply to typical ``pip`` usage where precompiled 'wheels' get downloaded and installed
+  and which can therefore be used on any architecture.
+
+Conda environments for R
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+For R, you need both the ``jupyter_client`` and the ``irkernel`` Conda packages installed.
+With the following command you can create the kernel::
 
       Rscript -e 'IRkernel::installspec(name="<env_name>", displayname="<kernel_name>")'
 
-Once the kernel is created, you will see it in the 'Launcher' menu. You can now start working in your own customized environment.
+Once the kernel is created, you will see it in the 'Launcher' menu.
+You can now start working in your own customized environment.
 
-For more general information concerning JupyterLab, go to their `official documentation <https://docs.jupyter.org/en/latest/>`__.
+For more general information, please refer to the `official JupyterLab documentation`_.
 
 **Remarks:**
 
-- The start location is `$VSC_DATA`
-- At the moment, we do not support installing extensions in Jupyter Lab
+- The top-level notebook directory is by default ``$VSC_DATA``.
+- At the moment, we do not support installing extensions in JupyterLab.
+
+.. note::
+
+   User kernels are stored by default in ``${VSC_HOME}/.local/share/jupyter/kernels``.
+   To override this and store your kernel specifications in a non-default location,
+   you may drop the following line in your ``${VSC_HOME}/.bashrc``::
+
+      export XDG_DATA_HOME=${VSC_DATA}/.local/share
+
+   If you have created multiple kernels (for Python and R), you find the corresponding
+   directories under ``${XDG_DATA_HOME}/jupyter/kernels``.
```

Comment:
I found this to be a bit unclear and I think this would improve it:
```suggestion
   When the ``${XDG_DATA_HOME}`` variable is set, subsequent kernel installations
   (for both Python and R) will reside in ``${XDG_DATA_HOME}/jupyter/kernels``.
```